pkg/oauth/oauthclient.go

```diff
@@ -0,0 +1,184 @@
+	glog.V(100).Infof("Pulling existing OAuthClient %s", name)
+
+	if apiClient == nil {
+		return nil, fmt.Errorf("apiClient cannot be nil")
+	}
+
+	builder := OAuthClientBuilder{
+		apiClient: apiClient.Client,
+		Definition: &oauthv1.OAuthClient{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The OAuthClient 'name' is empty")
+
+		return nil, fmt.Errorf("error: OAuthClient 'name' cannot be empty")
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("error: OAuthClient object %s not found", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get returns OAuthClient if found.
+func (builder *OAuthClientBuilder) Get() (*oauthv1.OAuthClient, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Fetching existing OAuthClient with name %s from cluster", builder.Definition.Name)
+
+	oauthClient := &oauthv1.OAuthClient{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, oauthClient)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return oauthClient, nil
+}
+
+// Create creates an OAuthClient object on the cluster from a builder.
+func (builder *OAuthClientBuilder) Create() (*OAuthClientBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the OAuthClient %s", builder.Definition.Name)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Exists checks whether the given OAuthClient exists.
+func (builder *OAuthClientBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if OAuthClientBuilder %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Update renovates a builder in the cluster and stores the created object in struct.
+func (builder *OAuthClientBuilder) Update() (*OAuthClientBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating the OAuthClient %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("error: OAuthClient object %s does not exist", builder.Definition.Name)
+	}
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+
+	if err == nil {
+		builder.Object = builder.Definition
+	}
+
+	return builder, nil
+}
+
+// Delete removes a OAuthClient from the cluster.
+func (builder *OAuthClientBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the OAuthClient %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return nil
```

Comment:
```suggestion
               glog.V(100).Infof("OAuthClient %s cannot be deleted"+
			" because it does not exist",
			builder.Definition.Name)
	
	        builder.Object = nil
	        
		return nil
```